village/src/components/SmurfForm.js

```diff
@@ -13,6 +13,14 @@ class SmurfForm extends Component {
   addSmurf = event => {
     event.preventDefault();
     // add code to create the smurf using the api
+    this.newSmurf = {
+      name: this.state.name,
+      age: this.state.age,
+      height: this.state.height,
+      id: this.props.smurfs.length + 1
```

Comment:
I don't think you need to manually add the id here because I think it is doing it on the server for you